activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -620,7 +620,7 @@ def extract_default_function(default_value, default)
         end
 
         def has_default_function?(default_value, default)
-          !default_value && %r{\w+\(.*\)|\(.*\)::\w+|CURRENT_DATE|CURRENT_TIMESTAMP}.match?(default)
+          !default.nil? && !default_value && %r{\w+\(.*\)|\(.*\)::\w+|CURRENT_DATE|CURRENT_TIMESTAMP}.match?(default)
```

Comment:
Maybe this to keep closer to the original code and avoid double negation:
```suggestion
          !default_value && default && %r{\w+\(.*\)|\(.*\)::\w+|CURRENT_DATE|CURRENT_TIMESTAMP}.match?(default)
```